web/src/api/getState.ts

```diff
@@ -22,32 +32,52 @@ const getState = async (timeAverage: string): Promise<GridState> => {
 };
 
 const useGetState = (): UseQueryResult<GridState> => {
-  const { urlTimeAverage } = useParams<RouteParameters>();
+  const queryClient = useQueryClient();
+  const { urlTimeAverage, urlDatetime } = useParams<RouteParameters>();
+  const isHourly = urlTimeAverage
+    ? URL_TO_TIME_AVERAGE[urlTimeAverage] === TimeAverages.HOURLY
+    : false;
+
+  const shouldUseLastHour = isHourly && !urlDatetime;
   const timeAverage = urlTimeAverage
     ? URL_TO_TIME_AVERAGE[urlTimeAverage]
     : TimeAverages.HOURLY;
-  const isHourly = urlTimeAverage ? timeAverage === TimeAverages.HOURLY : false;
-
-  // First fetch last hour only
-  const last_hour = useQuery<GridState>({
+  console.log('shouldUseLastHour', shouldUseLastHour, isHourly, urlTimeAverage);
+  const lastHourQuery = useQuery<GridState>({
     queryKey: [QUERY_KEYS.STATE, { aggregate: 'last_hour' }],
-    queryFn: async () => getState('last_hour'),
-    enabled: isHourly,
+    queryFn: () => getState('last_hour'),
+    enabled: shouldUseLastHour,
   });
 
-  const hourZeroWasSuccessful = Boolean(last_hour.isLoading === false && last_hour.data);
+  const fullDataQuery = useQuery<GridState>({
+    queryKey: [
+      QUERY_KEYS.STATE,
+      {
+        aggregate: timeAverage,
+        targetDatetime: urlDatetime,
+      },
+    ],
+    queryFn: () => getState(timeAverage, urlDatetime),
+    enabled: !shouldUseLastHour || (shouldUseLastHour && lastHourQuery.isSuccess),
```

Comment:
I think that the second reference to `shouldUseLastHour` is redundant. It should be true if the first branch didn't short circuit